### ISSUE TITLE
Samples in authentication doc page inconsistent

### ISSUE THREAD

**Issue by letmaik** [tsoa-community/docs#15]
https://tsoa-community.github.io/docs/authentication.html

In the first section, two `securityDefinitions` are defined: `api_key` and `tsoa_auth`, however, in the following, `api_token` and `jwt` are used. This is a bit confusing and should probably be aligned.

**Comment by Roman991** [comment]
#17

**Comment by cwilso03** [comment]
Even with the change in #17, there is still an inconsistency in the documentation.  The `spec.securityDefinitions.tsoa_auth` defintion isn't referenced in the example authentication middleware function.  And the `jwt` `securityName` referenced in the middleware function isn't declared in `spec.securityDefinitions`.

I'm unsure how a JWT auth scheme should now be configured in `spec.securityDefinitions` in the `tsoa.json` file.  When I first configured TSOA in my app back in early 2020, I used the following:

```
{
  "spec": {
    "securityDefinitions": {
      "jwt": {
        "bearerFormat": "JWT",
        "scheme": "bearer",
        "type": "http"
      }
    }
  }
}
```

This was based on the OpenAPI v3 instructions for bearer authentication (see https://swagger.io/docs/specification/authentication/bearer-authentication/), and it *does* still work.  But, I see now that I get lint warnings in my `tsoa.json` about the values of `jwt.scheme` ("Value should be one of: 'basic'") and `jwt.type` ("Value should be one of: 'basic', 'apiKey', 'oauth2'").
